addons/website/models/ir_ui_view.py

```diff
@@ -311,14 +322,19 @@ def _get_original_view(self):
         domain = [('key', '=', self.key), ('model_data_id', '!=', None)]
         return self.with_context(active_test=False).search(domain, limit=1)  # Useless limit has multiple xmlid should not be possible
 
-    def handle_visibility(self, do_raise=True):
+    def _handle_visibility(self, do_raise=True):
         """ Check the visibility set on the main view and raise 403 if you should not have access.
             Order is: Public, Connected, Has group, Password
 
             It only check the visibility on the main content, others views called stay available in rpc.
         """
         error = False
-        self = self.sudo()
+
+        try:
+            self.visibility  # avoid useless sudo()
```

Comment:
Just to avoid a useless sql query.
If it is to add a new query, I add sudo in all case.

But yes, I could handle More specific error, like ~ AccessError